### PR TITLE
Support 1.20.6

### PR DESCRIPTION
**Pull request by EpicPlayerA10**
### Changes
- Added required adapters for 1.20.6 due to internals breaking changes
- Restructured a bit `PlayerHeadAdapter`s
- Added paper-specific adapters too. This change will make dough future-proof on paper.
- Added exclusions in `pom.xml` in module `dough-protection` because dough couldn't be compiled. This is because one of dependencies provided very old version of bukkit which broke compilation
- In main `pom.xml` changed from spigot to paper to be able to use paper features in adapters

### CODE REVIEW

**Comment by ybw0014** [comment]
ItemUtils.damageItem needs an update as well, the unbreaking enchantment enum name is renamed.

**Review by DexterSK — APPROVED** [review]

**Comment by ybw0014** [comment]
@DexterSK making noises cannot let this PR get merged faster

**Comment by EpicPlayerA10** [comment]
> ItemUtils.damageItem needs an update as well, the unbreaking enchantment enum name is renamed.

We don't need to do this as from what i see `dough` is not updating minecraft version in the pom.xml:
https://github.com/Slimefun/dough/blob/baf2d79f6209b928209721981df0487f17410ac1/pom.xml#L287-L292
And spigot already auto renames these fields for plugins that are built for older minecraft versions: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/legacy/FieldRename.java . So no error will be thrown.

**Comment by mcchampions** [comment]
This pr conflicts with #219.

**Review by IvanMishin1 — APPROVED** [review]

**Comment by md5sha256** [comment]
> > ItemUtils.damageItem needs an update as well, the unbreaking enchantment enum name is renamed.
> 
> We don't need to do this as from what i see `dough` is not updating minecraft version in the pom.xml:
> 
> https://github.com/Slimefun/dough/blob/baf2d79f6209b928209721981df0487f17410ac1/pom.xml#L287-L292
> 
> 
> And spigot already auto renames these fields for plugins that are built for older minecraft versions: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/legacy/FieldRename.java . So no error will be thrown.

This does need to be updated as otherwise dough cannot compile if the consumer is compiling against 1.20.5+. The suggested fix is to refactor it to use the bukkit `Registry.ENCHANTMENT` api and grab it via the minecraft namespaced key.

**Comment by EpicPlayerA10** [comment]
Done

**Review by mcchampions — APPROVED** [review]

File: dough-skins/src/main/java/io/github/bakedlibs/dough/skins/nms/PlayerHeadAdapterPaper.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public void setGameProfile(Block block, GameProfile profile, boolean sendBlockUpdate) {
+        BlockState state = block.getState();
+        if (!(state instanceof Skull)) return;
+
+        Skull skull = (Skull) state;
+
+        Property property = profile.getProperties().get("textures").iterator().next();
+
+        PlayerProfile paperPlayerProfile = Bukkit.createProfile(profile.getId(), profile.getName());
+        paperPlayerProfile.setProperty(new ProfileProperty(property.name(), property.value(), property.signature()));
```
[the remark this answers is not in the document]
Comment (mcchampions) [reply dough-skins/src/main/java/io/github/bakedlibs/dough/skins/nms/PlayerHeadAdapterPaper.java:27]: I should not operate on mobile web pages

Comment (mcchampions) [inline dough-skins/src/main/java/io/github/bakedlibs/dough/skins/nms/PlayerHeadAdapterPaper.java:27]: RecordJava started using records in Java 16, while Authlib didn't use records until much later. As of the last version of Paper 1.18.2 that depended on Authlib, Property was still a class. Therefore, you might need to use reflection to call the getter method of an older version of Property.

**Review by mcchampions — CHANGES_REQUESTED** [review]

**Review by ybw0014 — CHANGES_REQUESTED** [review]
Based on the result server testing by boomer:

File: dough-items/src/main/java/io/github/bakedlibs/dough/items/nms/ItemNameAdapterPaper.java
Comment (ybw0014) [inline dough-items/src/main/java/io/github/bakedlibs/dough/items/nms/ItemNameAdapterPaper.java:1]: Adventure isn't bundle in paper until 1.19.2/.4.

**Review by ybw0014 — CHANGES_REQUESTED** [review]

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**